implants/lib/eldritch/src/pivot/arp_scan_impl.rs

```diff
@@ -1,5 +1,199 @@
+                            interface: interface.name.clone(),
+                        });
+                        break;
+                    }
+                }
+                Err(e) => {
+                    if e.kind() == std::io::ErrorKind::TimedOut {
+                        continue;
+                    }
+                    println!("Error receiving packet: {}", e);
+                    return;
+                }
+            }
+        }
+    }
+}
+
+pub fn arp_scan(starlark_heap: &Heap, target_cidrs: Vec<String>) -> Result<Vec<Dict>> {
+    let target_cidrs = target_cidrs
+        .iter()
+        .map(|cidr| {
+            let (addr, prefix) = cidr.split_at(cidr.find('/').unwrap());
+            let addr = match Ipv4Addr::from_str(addr) {
+                Ok(addr) => addr,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid IPv4 address: {}", addr));
+                }
+            };
+            let prefix: u8 = match prefix[1..].parse() {
+                Ok(prefix) => prefix,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid CIDR prefix: {}", prefix));
+                }
+            };
+            let network = match Ipv4Network::new(addr, prefix) {
+                Ok(network) => network,
+                Err(_) => {
+                    return Err(anyhow::anyhow!("Invalid CIDR: {}", cidr));
+                }
+            };
+            Ok(network)
+        })
+        .collect::<Result<Vec<Ipv4Network>>>();
+    match target_cidrs {
+        Ok(target_cidrs) => {
+            let listener_out: Arc<Mutex<HashMap<Ipv4Addr, Option<ArpResponse>>>> =
+                Arc::new(Mutex::new(HashMap::new()));
+            for target_cidr in target_cidrs {
+                for ip in target_cidr.iter() {
+                    listener_out.lock().unwrap().insert(ip, None);
+                }
+            }
+            let interfaces = datalink::interfaces();
+            for interface in interfaces {
+                let inner_out = listener_out.clone();
+                let interface = interface.clone();
+                let thread = std::thread::spawn(move || {
+                    start_listener(interface, inner_out);
+                });
+                thread.join().unwrap();
+            }
+            let mut out: Vec<Dict> = Vec::new();
+            let final_listener_output = listener_out.lock().unwrap().clone();
+            for (ipaddr, res) in final_listener_output {
+                if let Some(res) = res {
+                    let hit_small_map = SmallMap::new();
+                    let mut hit_dict = Dict::new(hit_small_map);
+                    let ipaddr_value = starlark_heap.alloc_str(&ipaddr.to_string());
+                    let source_mac_value = starlark_heap.alloc_str(&res.source_mac.to_string());
+                    let interface_value = starlark_heap.alloc_str(&res.interface.to_string());
+                    hit_dict.insert_hashed(
```

Comment:
Please use lower case variables it’s easier to remember for users if every variable is lower case.